### ISSUE TITLE
Object.dirname is deprecated.

### ISSUE THREAD

**Issue by mxxcon** [mschuchard/linter-ansible-linting#29]
Argument to `path.dirname` must be a string
```
Object.dirname (/Applications/Atom.app/Contents/Resources/app.asar/src/electron-shims.js:9:10)
Object.lint (/Users/mxx/.atom/packages/linter-ansible-linting/lib/main.js:227:127)
a.factory (/Users/mxx/.atom/packages/atom-ide-ui/modules/atom-ide-ui/pkg/atom-ide-diagnostics/lib/services/LinterAdapter.js:309:42)
a._callFactory (/Users/mxx/.atom/packages/atom-ide-ui/node_modules/rxjs/bundles/Rx.min.js:107:23)
a.tryDefer (/Users/mxx/.atom/packages/atom-ide-ui/node_modules/rxjs/bundles/Rx.min.js:106:442)
new a (/Users/mxx/.atom/packages/atom-ide-ui/node_modules/rxjs/bundles/Rx.min.js:106:383)
```

I just got this deprecation warning. Atom 1.32.0 on OSX

**Comment by mschuchard** [comment]
I am running 1.31.2. I am not seeing this warning in the CI for atom 1.32.0-beta3. When my local install updates I will probably run into this too if it exists.

Ping me in a week or so if I do not update the issue by then.

**Comment by Arcanemagus** [comment]
This is likely due to linting being requested on an unsaved file or on a remote file (such as Nuclide or Teletype). You'll want to add a check that the path is defined, and doesn't include `://` right after this line:
https://github.com/mschuchard/linter-ansible-linting/blob/e201603dd623838171ad703cd998c6f6663132f5/lib/main.js#L163

You should probably also put a check on the `activeEditor` itself to ensure it's really a `TextEditor` using [`atom.workspace.isTextEditor()`](https://atom.io/docs/api/v1.32.0/Workspace#instance-isTextEditor) 😉. It _shouldn't_ be possible, but sometimes during initialization you can get a request with a TextEditor that is removed during the deserialization process.

**Comment by Arcanemagus** [comment]
> Is it possible to lint on a remote file

That depends on the actual linter being ran: _Most_ of them require some knowledge of the rest of the files, however if it can run on _just_ the contents of the file I think you should be fine returning results.

**Comment by mxxcon** [comment]
Yes, I think that was the problem. I just opened a new tab and copy/pasted some code in it from a remote server.

**Comment by mschuchard** [comment]
Addressed in 1e70a1a. Due to ansible-lint requirements around pathing for include and role linting from playbooks, reliable linting cannot occur from stdin and so I am forced to block linting on nonexistent playbook files for the time being. A useful info message is now displayed to assist.
